### PR TITLE
test: Add default deny egress test for net work policy.

### PR DESCRIPTION
**Pull request by csfmomo**
Add egress rule e2e test for network policy. 

**Reason for Change**:
E2E test didn't cover egress network policy. Add a default deny all egress policy to the E2E test.

**Issue Fixed**:
Lack of egress network policy E2E test.

**Requirements**:

- [ X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [X ] tested upgrade from previous version

### CODE REVIEW

**Comment by csfmomo** [comment]
Reopen it to see whether trigger auto check.

**Comment by jackfrancis** [comment]
Thanks @csfmomo! I'm testing this additional test coverage against all of our NetworkPolicy-enabled cluster configurations.

**Comment by jackfrancis** [comment]
This test failed, here's output:

```
$ k describe pod network-policy-kubernetes-eastus-38698-382588-6f59c75567-dr5m8 -n development completed in 451.427423ms
 2020/03/11 22:25:34 
 Name:         network-policy-kubernetes-eastus-38698-382588-6f59c75567-dr5m8
 Namespace:    development
 Priority:     0
 Node:         k8s-agentpool1-26536814-vmss000001/10.240.0.35
 Start Time:   Wed, 11 Mar 2020 22:20:23 +0000
 Labels:       app=webapp
               pod-template-hash=6f59c75567
               role=any
 Annotations:  kubernetes.io/psp: privileged
 Status:       Running
 IP:           10.240.0.49
 IPs:
   IP:           10.240.0.49
 Controlled By:  ReplicaSet/network-policy-kubernetes-eastus-38698-382588-6f59c75567
 Containers:
   network-policy-kubernetes-eastus-38698-382588:
     Container ID:   docker://06fea134cd05fd1e9cbbf393462ece22cea078e82839f63b9b84f95605e6cb95
     Image:          library/nginx:latest
     Image ID:       docker-pullable://nginx@sha256:2539d4344dd18e1df02be842ffc435f8e1f699cfc55516e2cf2cb16b7a9aea0b
     Port:           <none>
     Host Port:      <none>
     State:          Running
       Started:      Wed, 11 Mar 2020 22:20:25 +0000
     Ready:          True
     Restart Count:  0
     Requests:
       cpu:        10m
       memory:     10M
     Environment:  <none>
     Mounts:
       /var/run/secrets/kubernetes.io/serviceaccount from default-token-27mcp (ro)
 Conditions:
   Type              Status
   Initialized       True 
   Ready             True 
   ContainersReady   True 
   PodScheduled      True 
 Volumes:
   default-token-27mcp:
     Type:        Secret (a volume populated by a Secret)
     SecretName:  default-token-27mcp
     Optional:    false
 QoS Class:       Burstable
 Node-Selectors:  beta.kubernetes.io/os=linux
 Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                  node.kubernetes.io/unreachable:NoExecute for 300s
 Events:
   Type    Reason     Age        From                                         Message
   ----    ------     ----       ----                                         -------
   Normal  Scheduled  <unknown>  default-scheduler                            Successfully assigned development/network-policy-kubernetes-eastus-38698-382588-6f59c75567-dr5m8 to k8s-agentpool1-26536814-vmss000001
   Normal  Pulling    5m9s       kubelet, k8s-agentpool1-26536814-vmss000001  Pulling image "library/nginx:latest"
   Normal  Pulled     5m9s       kubelet, k8s-agentpool1-26536814-vmss000001  Successfully pulled image "library/nginx:latest"
   Normal  Created    5m9s       kubelet, k8s-agentpool1-26536814-vmss000001  Created container network-policy-kubernetes-eastus-38698-382588
   Normal  Started    5m9s       kubelet, k8s-agentpool1-26536814-vmss000001  Started container network-policy-kubernetes-eastus-38698-382588
 
 2020/03/11 22:25:35 Error trying to create NetworkPolicy backend-deny-egress:error: error validating "workloads/policies/backend-policy-deny-egress.yaml": error validating data: ValidationError(NetworkPolicy.spec.podSelector): unknown field "policyTypes" in io.k8s.apimachinery.pkg.apis.meta.v1.LabelSelector; if you choose to ignore these errors, turn validation off with --validate=false
```

**Comment by csfmomo** [comment]
> /hold

Hi @jackfrancis , thanks for reviewing and sharing test output for it. I repro this failure in local and fixed it in my second revision. The test should pass now.

File: test/e2e/kubernetes/kubernetes_test.go [hunk 1]
```diff
@@ -1621,6 +1621,27 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
...
 				networkpolicy.DeleteNetworkPolicy(nwpolicyName, namespace)
 
+				By("Applying a network policy to deny egress access in development namespace")
+				nwpolicyName, namespace = "backend-deny-egress", nsDev
+				err = networkpolicy.CreateNetworkPolicyFromFile(filepath.Join(PolicyDir, "backend-policy-deny-egress.yaml"), nwpolicyName, namespace)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Ensuring we no longer have egress access from the network-policy pods to backend pods")
+				for _, backendPod := range backendPods {
+					pass, err = pl.ValidateCurlConnection(backendPod.Status.PodIP, 30*time.Second, validateNetworkPolicyTimeout)
```
Comment (jaer-tsun) [inline test/e2e/kubernetes/kubernetes_test.go]: I think we can shorten the timeout, 30s is quite long

**Pull request merged**